android/app/src/main/java/org/fossasia/openevent/viewmodels/auth/LoginActivityViewModel.java

```diff
@@ -0,0 +1,66 @@
+package org.fossasia.openevent.viewmodels.auth;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import android.util.Pair;
+
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.auth.Login;
+import org.fossasia.openevent.utils.Utils;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class LoginActivityViewModel extends ViewModel {
+    public static final int VALID = 0;
+    public static final int EMPTY = 1;
+    public static final int INVALID = 2;
+
+    private MutableLiveData<Pair<Integer,String>> loginResponse;
+    private final CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    public int validateEmail(String email) {
+        if (Utils.isEmpty(email)) {
+            return EMPTY;
+        } else if (!Utils.isEmailValid(email)) {
+            return INVALID;
+        }
+        return VALID;
+    }
+
+    public int validatePassword(String password) {
+        if (Utils.isEmpty(password)) {
+            return EMPTY;
+        } else if (!Utils.isPasswordValid(password)) {
+            return INVALID;
+        }
+        return VALID;
+    }
+
+    public LiveData<Pair<Integer,String>> loginUser(String email, String password) {
+        if (loginResponse == null) {
+            loginResponse = new MutableLiveData<>();
+        }
+        compositeDisposable.add(APIClient.getOpenEventAPI().login(new Login(email, password))
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(response -> {
+                            Timber.d("Saved token and logged in successfully");
+                            loginResponse.setValue(new Pair<>(SignUpActivityViewModel.VALID, response.getAccessToken()));
```

Comment:
Let's not use Pair. Create a new class if you need to, but any Android specific class is completely forbidden

Reply:
If used in other PRs, please correct

Reply:
> but any Android specific class is completely forbidden

Isn't Pair<> a java specific class ? I don't think it's Android specific class.

Reply:
There's no pair in Java. `import android.util.Pair;` The more you know :smile:

Reply:
There should be one :stuck_out_tongue_winking_eye: